dask/array/blockwise.py

```diff
@@ -197,7 +199,26 @@ def blockwise(func, out_ind, *args, **kwargs):
                         "adjust_chunks values must be callable, int, or tuple")
     chunks = tuple(chunks)
 
-    return Array(graph, out, chunks, dtype=dtype)
+    # Attempt to construct meta, fallback to Array's dtype constructor if it's
+    # a basic type
+    try:
+        arrays = args[::2]
+        ndims = [a.ndim if hasattr(a, 'ndim') else 0 for a in arrays]
+        args_meta = [arg._meta if hasattr(arg, '_meta') else arg
+                     for arg, nd in zip(arrays, ndims)]
+        kwargs_meta = {k: v._meta if hasattr(v, '_meta') else v for k, v in kwargs.items()}
+        try:
+            meta = func(*args_meta, **kwargs_meta)
+        except:
+            meta = func(*args_meta)
+
+        if np.isscalar(meta):
+            ndim = max([a.ndim for a in arrays])
+            meta = arrays[0]._meta[tuple(slice(0, 0, None)
+                                   for _ in range(ndim))]
+        return Array(graph, out, chunks, meta=meta.astype(dtype))
+    except:
+        return Array(graph, out, chunks, dtype=dtype)
```

Comment:
I'm a little nervous that we are adding a few bare `except`s here for a couple of reasons.

1. Bare `except`s will catch things like `KeyboardInterrupt`, which we shouldn't catch.
2. It seems we don't have a good handle on what kind of errors are coming or why.

TBH 2 concerns me more than 1.

Though I haven't played with this code yet. So I may be overly concerned about something that may be in fact easy to fix. More information on why these are needed would be good.

Reply:
No, you're not overly concerned. This wasn't meant to be the merge-ready code, I only catched bare exceptions to ease my life while I was testing all cases. The exceptions we catch should _definitely_ be evaluated properly, which I still intend to do, but first I was waiting for criticism regarding the idea.

This last exception in particular is a difficult one, I've identified that on the `try` block, several different ones may occur, so far I've identified:

* `ValueError`
* `AttributeError`
* `TypeError`
* `IndexError`
* `ZeroDivisionError`

And to be even worse, other exceptions might happen during the exception handling, I don't currently know how to best deal with them here.

Reply:
In the worst case scenario you should still `except Exception:` rather than `except:`.  This excludes some more exotic errors like `KeyboardInterrupt`s